libyul/optimiser/Suite.cpp

```diff
@@ -164,6 +167,7 @@ void OptimiserSuite::run(
 			RedundantAssignEliminator::run(*_dialect, ast);
 			UnusedPruner::runUntilStabilised(*_dialect, ast, reservedIdentifiers);
 			CommonSubexpressionEliminator{*_dialect}(ast);
+			DeadCodeEliminator{}(ast);
```

Comment:
Actually the most likely place where this can eliminate substantial code is right after the full inliner (at the end, i.e. after the block flattener).